### PR TITLE
Stop hitting master/state.json in mesos collector

### PR DESCRIPTION
**Pull request by sagar8192**
Please see PAASTA-13169 for more context.

Summary:
The fullerite mesos collector was using /master/state.json endpoint to figure out the ip address of the elected mesos master. This is an extremely expensive call which returns a json blob of mesos masters internal state. There is another endpoint 'metrics/snapshot' which can be used to check if the master we are using is the elected leader or not. This endpoint is much lighter than 'master/state.json'.

The change I made:
Removed mesos leader utility and switched to 'metrics/snapshot' to get metrics.

Testing:
Manually tested.
Unit tests are passing.
Added a test to make sure that my change works.

### CODE REVIEW

**Comment by drolando** [comment]
lgtm, but I'd wait for Avadhut's review

File: src/fullerite/collector/mesos.go [hunk 1]
```diff
@@ -163,6 +152,14 @@ func (m *MesosStats) getMetrics(ip string) map[string]float64 {
 		return nil
 	}
 
+    // Check if it the elected master or not.
+    if snapshot["master.elected"] == 1 {
+        m.log.Warning("This is the elected leader!")
```
Comment (solarkennedy) [inline src/fullerite/collector/mesos.go]: I don't think you need to warn all the time, in either case. Maybe Debug() ?

File: src/fullerite/collector/mesos.go [hunk 2]
```diff
@@ -163,6 +152,14 @@ func (m *MesosStats) getMetrics(ip string) map[string]float64 {
...
 
+    // Check if it the elected master or not.
+    if snapshot["master.elected"] == 1 {
+        m.log.Debug("This is the elected leader!")
+    } else {
+        m.log.Debug("This is not the leader!")
+        return make(map[string]float64)
+    }
+
 	return snapshot
```
Comment (solarkennedy) [inline src/fullerite/collector/mesos.go]: Will this return no metrics at all? Or will it send all 0s?
The previous behavior we didn't even send anything in the `Collect` function, but now we will always be sending "something". What is that something in the non-leader case?

Comment (sagar8192) [reply src/fullerite/collector/mesos.go]: Since we are returning an empty map in the non-leader case, this should not emit any metrics.

**Comment by sagar8192** [comment]
Going to wait for @avadhutp's approval before merging this.

**Comment by avadhutp** [comment]
lgtm; ship-it :shipit:

**Pull request merged**